### PR TITLE
[14.0][MIG] partner_statement: Migration to 14.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 14.0

Please @chienandalu and @pedrobaeza can you review it?

@Tecnativa TT30008

### CODE REVIEW

File: partner_statement/__manifest__.py [hunk 1]
```diff
@@ -3,16 +3,17 @@
 
 {
     "name": "Partner Statement",
-    "version": "13.0.1.2.0",
+    "version": "14.0.1.0.0",
     "category": "Accounting & Finance",
     "summary": "OCA Financial Reports",
-    "author": "ForgeFlow, Odoo Community Association (OCA)",
+    "author": "ForgeFlow, Tecnativa, Odoo Community Association (OCA)",
```
Comment (pedrobaeza) [inline partner_statement/__manifest__.py:9]: No co-authorship should be granted in this simple migration

File: partner_statement/tests/test_outstanding_statement.py [hunk 2]
```diff
@@ -51,14 +51,13 @@ def test_customer_outstanding_statement(self):
 
         statement = wiz_id.button_export_pdf()
 
-        self.assertDictContainsSubset(
+        self.assertGreaterEqual(
```
Comment (pedrobaeza) [inline partner_statement/tests/test_outstanding_statement.py:54]: Please don't change this, as it doesn't work the same way.

File: partner_statement/tests/test_activity_statement.py [hunk 3]
```diff
@@ -55,14 +55,13 @@ def test_customer_activity_statement(self):
 
         statement = wiz_id.button_export_pdf()
 
-        self.assertDictContainsSubset(
+        self.assertGreaterEqual(
```
Comment (pedrobaeza) [inline partner_statement/tests/test_activity_statement.py:58]: Please don't change this, as it doesn't work the same way.

File: partner_statement/report/report_statement_common.py [hunk 4]
```diff
@@ -66,7 +69,7 @@ def _show_buckets_sql_q1(self, partners, date_end, account_type):
                 WHERE l2.date <= %(date_end)s
             ) as pc ON pc.credit_move_id = l.id
             WHERE l.partner_id IN %(partners)s
-                                AND l.account_internal_type = %(account_type)s
+                                AND l_account_type.type = %(account_type)s
```
Comment (pedrobaeza) [inline partner_statement/report/report_statement_common.py:72]: Why adding extra table? I think the field is still available, at least on `account_account`, isn't it?

File: partner_statement/report/outstanding_statement.py [hunk 5]
```diff
@@ -57,7 +60,7 @@ def _display_lines_sql_q1(self, partners, date_end, account_type):
                 WHERE l2.date <= %(date_end)s
             ) as pc ON pc.credit_move_id = l.id
             WHERE l.partner_id IN %(partners)s
-                                AND l.account_internal_type = %(account_type)s
+                                AND l_account_type.type = %(account_type)s
```
Comment (pedrobaeza) [inline partner_statement/report/outstanding_statement.py:63]: The same as the other query

File: partner_statement/report/activity_statement.py [hunk 6]
```diff
@@ -107,9 +110,11 @@ def _display_lines_sql_q1(self, partners, date_start, date_end, account_type):
                 END as date_maturity
             FROM account_move_line l
             JOIN account_move m ON (l.move_id = m.id)
+            JOIN account_account l_account ON l.account_id = l_account.id
+            JOIN account_account_type l_account_type ON l_account.user_type_id = l_account_type.id
             JOIN account_journal aj ON (l.journal_id = aj.id)
             WHERE l.partner_id IN %(partners)s
-                AND l.account_internal_type = %(account_type)s
+                AND l_account_type.type = %(account_type)s
```
Comment (pedrobaeza) [inline partner_statement/report/activity_statement.py:117]: The same as the other query

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
And please squash together the 3 commits for Spanish (Argentina) translation.

**Comment by gdgellatly** [comment]
Why do this and ignore the open pull request?

**Comment by pedrobaeza** [comment]
We didn't know about the other. Is it in better condition then this one? Can you point to it?

**Comment by gdgellatly** [comment]
https://github.com/OCA/account-financial-reporting/pull/762 As for better condition, well it has a lot more manual testing but still I feel we haven't quite got the currency part correct. There are some minor fixes needed as well but for single currency it works well.

We have a load of downstream dependent modules which automate a lot of the reports so as those are migrated this kind of gets extra tested.

**Comment by pedrobaeza** [comment]
Closing in favor of #762

**Pull request closed without merging**